### PR TITLE
Ridable carps from bee [PORT]

### PR DESCRIPTION
**Pull request by dwasint**
About The Pull Request

This is not an exact port! This is an interpretation of someone's suggestion, no external code was referenced, or needs attributing.
https://github.com/BeeStation/BeeStation-Hornet/pull/7174
Introduces several types of lassos.

    Basic
    Primal
    Antag
    Debug
    These lassos can be used to tame animals. Specifically, the basic lasso can tame space-carp & cows, this type can be crafted. The primal lasso can also be crafted, exclusively by ash-walkers, can be used to tame certain lavaland fauna. The antag lasso can tame anything outside of a specified blacklist, bubblegum & other bosses, can be bought from any uplink for 3TC. The debug lasso tames anything.
    Tamed animals can be rode & given commands through speech and radial menus(alt-click).
    Radial menu icons are procedurally generated, based on the animal's icon.

Why It's Good For The Game

Unique item, free content.
Testing Photographs and Procedure
Screenshots&Videos
Coder sprites - Subject to change
Changelog

🆑
add: Adds lasso items
add: Adds tamed riding component subtype, toggles AI & sometimes spacewalking
add: Adds antag lasso to syndicate uplink
add: Add new AI for tamed animals
add: Add tamed component
/🆑

### CODE REVIEW

**Comment by Salted-Zen** [comment]
Maybe its a typo, but it says that the lasso's for lavaland will only be craftable by ash walkers, but perhapse miners should have access too, maybe as either a point reward, or just make it a tribal craft for anyone who can get materials from lavaland fauna.

File: code/datums/beam.dm [hunk 1]
```diff
@@ -159,19 +159,7 @@
...
- * icon_state: What the beam's icon_state is. The datum effect isn't the ebeam object, it doesn't hold any icon and isn't type dependent.
- * icon: What the beam's icon file is. Don't change this, man. All beam icons should be in beam.dmi anyways.
- * maxdistance: how far the beam will go before stopping itself. Used mainly for two things: preventing lag if the beam may go in that direction and setting a range to abilities that use beams.
- * beam_type: The type of your custom beam. This is for adding other wacky stuff for your beam only. Most likely, you won't (and shouldn't) change it.
- */
-/atom/proc/Beam(atom/BeamTarget,icon_state="b_beam",icon='icons/effects/beam.dmi',time=INFINITY,maxdistance=INFINITY,beam_type=/obj/effect/ebeam, beam_color = null, beam_type=/obj/effect/ebeam,beam_sleep_time = 3)
-	var/datum/beam/newbeam = new(src,BeamTarget,icon,icon_state,time,maxdistance,beam_type, beam_color,beam_sleep_time)
+/atom/proc/Beam(atom/BeamTarget,icon_state="b_beam",icon='icons/effects/beam.dmi',time=50, maxdistance=10,beam_type=/obj/effect/ebeam,beam_sleep_time = 3)
+	var/datum/beam/newbeam = new(src,BeamTarget,icon,icon_state,time,maxdistance,beam_type,beam_sleep_time)
```
Comment (nednaZ) [inline code/datums/beam.dm:163]: You deleted a large amount of documentation.
Also, if this is related to your beam refactor it's not going to work.
I had already reported a major issue with it that was never fixed

Comment (dwasint) [reply code/datums/beam.dm]: reverted, no idea how this got in here my main guess is github desktop pulling changes from an old branch

**Comment by EvilDragonfiend** [comment]
don't forget https://github.com/BeeStation/BeeStation-Hornet/pull/7345

**Pull request closed without merging**